### ISSUE TITLE
Error in Mosaic Augmentation

### ISSUE THREAD

**Issue by developer0hye** [ultralytics/ultralytics#2609]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

# Command

```
yolo task=detect mode=train device=0,1 data=coco.yaml epochs=300 batch=64 project=exp8-1 cache=disk model=yolov8n.yaml name=yolov8n scale=0.9 mixup=0.15 copy_paste=0.3 close_mosaic=10
```

# Error
```
Starting training for 300 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/300      6.46G  3.564e+04      5.735  2.789e+04       1086        640:   1%|          | 16/1849 [01:20<1:24:27,  2.76s/it]Traceback (most recent call last):
  File "/root/.config/Ultralytics/DDP/_temp_ep6qv256139798880687184.py", line 6, in <module>
    trainer.train()
  File "/home/ultralytics/ultralytics/yolo/engine/trainer.py", line 191, in train
    self._do_train(world_size)
  File "/home/ultralytics/ultralytics/yolo/engine/trainer.py", line 322, in _do_train
    for i, batch in pbar:
  File "/home/ultralytics/ultralytics/yolo/data/build.py", line 38, in __iter__
    yield next(self.iterator)
  File "/opt/conda/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 628, in __next__
    data = self._next_data()
  File "/opt/conda/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 1333, in _next_data
    return self._process_data(data)
  File "/opt/conda/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 1359, in _process_data
    data.reraise()
  File "/opt/conda/lib/python3.10/site-packages/torch/_utils.py", line 543, in reraise
    raise exception
ValueError: Caught ValueError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/torch/utils/data/_utils/worker.py", line 302, in _worker_loop
    data = fetcher.fetch(index)
  File "/opt/conda/lib/python3.10/site-packages/torch/utils/data/_utils/fetch.py", line 58, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/opt/conda/lib/python3.10/site-packages/torch/utils/data/_utils/fetch.py", line 58, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/ultralytics/ultralytics/yolo/data/base.py", line 220, in __getitem__
    return self.transforms(self.get_label_info(index))
  File "/home/ultralytics/ultralytics/yolo/data/augment.py", line 56, in __call__
    data = t(data)
  File "/home/ultralytics/ultralytics/yolo/data/augment.py", line 56, in __call__
    data = t(data)
  File "/home/ultralytics/ultralytics/yolo/data/augment.py", line 104, in __call__
    labels = self._mix_transform(labels)
  File "/home/ultralytics/ultralytics/yolo/data/augment.py", line 149, in _mix_transform
    return self._mosaic4(labels) if self.n == 4 else self._mosaic9(labels)
  File "/home/ultralytics/ultralytics/yolo/data/augment.py", line 224, in _mosaic9
    img9[y1:y2, x1:x2] = img[y1 - padh:, x1 - padw:]  # img9[ymin:ymax, xmin:xmax]
ValueError: could not broadcast input array from shape (427,641,3) into shape (427,640,3)

```

### Environment

Ultralytics YOLOv8.0.100 🚀 Python-3.10.8 torch-1.13.1 CUDA:0 (NVIDIA GeForce RTX 3090, 24260MiB)
                                                       CUDA:1 (NVIDIA GeForce RTX 3090, 24260MiB)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by developer0hye** [comment]
@Laughing-q Hi! My version is Ultralytics YOLOv8.0.100.

**Comment by Laughing-q** [comment]
@developer0hye ok got it. We actually introduce a mosaic update in 8.0.100 which might be the cause of this issue, could you try the same training command with an old version like 8.0.99/98? I'd like to know if it's related to the new update of mosaic. Thanks!

**Comment by developer0hye** [comment]
@Laughing-q 

I am testing it with an old version 8.0.99 now! It is being handled well up to 4%.
It seems that l
![image](https://github.com/ultralytics/ultralytics/assets/35001605/f35dbfa4-e2c1-47aa-9806-9e211fd0632c)

**Comment by Laughing-q** [comment]
@developer0hye ok got it. Then I'll try to reproduce the issue and fix it. Thanks!

cc @glenn-jocher

**Comment by Laughing-q** [comment]
@developer0hye hi, the issue has been fixed in the latest version 8.0.101. :)

**Comment by developer0hye** [comment]
@Laughing-q What a quick fix! Thanks!

**Comment by glenn-jocher** [comment]
@developer0hye you're welcome, glad to hear that the issue has been resolved for you in YOLOv8 8.0.101. If you have any further concerns or questions, feel free to ask anytime.

**Comment by janardan-ds** [comment]
@glenn-jocher **Thanks for the quick fix!,** I was also stucked and rasied the issue but just after updatinh the latest verison 8.0.101 now it's working.
It's not yet updated in pip install ultralytics --> installing the previous verson which is throwing error error in mosaic augmentation.

**Comment by glenn-jocher** [comment]
@janardan-ds thank you for reaching out and letting us know that the issue has been resolved for you in YOLOv8 version 8.0.101. We always strive to quickly fix any issues that may arise and glad to hear that our efforts have helped. Regarding your note about pip install ultralytics: sometimes it takes a little time for the latest version to become available through pip. In the meantime, you can manually install the latest version from our GitHub repo. Let us know if you have any further questions or concerns.
